![logo-gt.png](attachment:logo-gt.png)

<div class="alert alert-block alert-info">
    <h1>BIOS 4150/BIOL 6150</h1>
    <h3>Instructor: Dr. King Jordan</h3>
    <p>Shivam Sharma (shivamsharma13@gatech.edu),</p>
    <p>Nilavrah Sensarma (nsensarma3@gatech.edu),</p> 
    <p>Bengy Gyimah-Asamoah, (bgyimaha3@gatech.edu)</p>
</p>
</div>

<div class="alert alert-block alert-warning">
    <h2>Project 1 (Data access & QC) starter notebook</h2>
    <h3>Deadline: 11:59PM, September 24rd, 2024</h3>
</div>

---

# **1. Your assigned 1000 Genomes individual**
### *Total Questions: 3*
### *Total Points: 3+3+4 = 10*


---

<div class="alert alert-block alert-warning">
    <h3>1.1 Does you group have an assigned individual ID from the 1000 Genomes Project?</h3>
    <p>Find the individual on the international genomes website and paste a screenshot below</p>
</div>

In [ ]:
##Add image here for 1.1


![AssignedIndividual.png](./AssignedIndividual.png)

<div class="alert alert-block alert-warning">
    <h3>1.2 What population does your assigned individual belong to?</h3>
    <p>Write about where the geographical location of where this individual lived at the time of recruitment </p>
</div>

In [ ]:
#Write about the individual and their ancestors here.

She was born and is from Barbados, as well as her past two ancestral generations (her grandparents) on both the maternal and paternal side. 

<div class="alert alert-block alert-warning">
    <h3>1.3 More about your individual's population?</h3>
    <p>Read more about this population and write about their recent historical  (100-500 years) ancestors.</p>
</div>

In [ ]:
#Write about the individual and their ancestors here.

She is of African Caribbean ancestry (which both sets of grandparents also identify with as they were also born in Barbados), meaning that her distant ancestors were originally from Africa, but recently are from the Caribbean. 

# **2. Exome sequencing**
### *Total Questions: 3*
### *Total Points: 5+5=10*


---

<div class="alert alert-block alert-warning">
    <h3>2.1 Does your assigned individual has exome sequencing data? If yes, then what is the SRR ID for that sequencing run?</h3>
    <p>Does you individual have multiple SRR IDs? In case they do, which one would you pick? (Hint: The sequencing information is coming from the same individual; but larger file size might be more preferable)</p>
</div>

In [ ]:
#Write about the SRR ID here.

The SRR ID is SRR393028. We chose this one because it is the only one with a paired sequence (SRR393028_1 and SRR393028_2).

<div class="alert alert-block alert-warning">
    <h3>2.2 Looking at just the exome sequencing of this individual, how many variants would you expect them to carry compared to the reference genome?</h3>
    <p>Keep in mind this is 1 individual, and learn about the expected number of variants from WGS, WES, and Genotyping</p>
</div>

According to a study, a person’s ~6 billion-nucleotide genome sequence will have around 5,000,000 SNVs, 600,000 insertion/deletion variants, and 25,000 structural variants. Our genome sequence has total 4611040 reads, so the expected number of variants would be 18444. The amount of data produced for analysis in WES is approximately 10 Gb (150x) per patient, whereas the amount of data produced in WGS is 120 Gb (35x).

# **3. Data access**
### *Total Questions: 3*
### *Total Points: 5+5 = 10*


---

<div class="alert alert-block alert-warning">
    <h3>3.1 Download the exome sequencing data for this individual and write the size of the files?</h3>
    <p>3.1.1 How many files do you have? Is the sequencing paired?</p>
    <p>3.1.2 How many reads are present in each file? </p>
</div>

In [ ]:
#Show your download command and ls the directory.

In [1]:
#Download the file.
!wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR393/SRR393028/SRR393028_1.fastq.gz
!wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR393/SRR393028/SRR393028_2.fastq.gz

--2024-09-21 18:43:26--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR393/SRR393028/SRR393028_1.fastq.gz
           => ‘SRR393028_1.fastq.gz’
Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.165
Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.165|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR393/SRR393028 ... done.
==> SIZE SRR393028_1.fastq.gz ... 51844801
==> PASV ... done.    ==> RETR SRR393028_1.fastq.gz ... done.
Length: 51844801 (49M) (unauthoritative)

SRR393028_1.fastq.g 100%[===================>]  49.44M  6.43MB/s    in 9.8s    

2024-09-21 18:43:37 (5.03 MB/s) - ‘SRR393028_1.fastq.gz’ saved [51844801]

--2024-09-21 18:43:37--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR393/SRR393028/SRR393028_2.fastq.gz
           => ‘SRR393028_2.fastq.gz’
Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.165
Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62

![proj1terminal.png](./proj1terminal.png)

In [3]:
#Write about the fastq file here.

We downloaded two files SRR39028_1.fastq.gz and SRR39028_2.fastq.gz. 

In SRR393028_1.fastq.gz, there are 51844801 reads in each file. In SRR393028_2.fastq.gz, there are 52663283 reads in each file. 

These are paired sequences.

In terminal, there is no .qz because the screenshot was taken after we unzipped the files.

In [2]:
#Have a quick look at the files downloaded.
!zcat ~/biol6150/ProjectSubmissions/Group8/Project1/SRR393028_1.fastq.gz | head 
!echo "----"
!zcat ~/biol6150/ProjectSubmissions/Group8/Project1/SRR393028_2.fastq.gz | head 

@SRR393028.1 1/1
CACAGACGGGTAGAACCTCAGTAATCCGAAAAGCCGGGGTCGACCGCCCCTTGCTTGCAGCCGGGCACTACAGGAC
+
????????????????????????????????????????????????????????????????????????????
@SRR393028.2 2/1
CATGTAACCTAATACCACGACCAGGCATGGGGGAAAGATTGGAGGAAAGTTGAGTGAGAGGATCAACTTCTCTGAC
+
????????????????????????????????????????????????????????????????????????????
@SRR393028.3 3/1
AGAACGGAGCAGCTGGTGATGTGTGGGCCCACCGGCCCCAGGCTCCTGTCTCCCCCCAGGTGTGTGGTGATGCCAG

gzip: stdout: Broken pipe
----
@SRR393028.1 1/2
TGCCACCACTCTAAGCAAGAGAGCCCTGCAGTTGCCCTAGTCGCCAGCAGGGGGCGCCCTGGCACAGCACCGTGAG
+
????????????????????????????????????????????????????????????????????????????
@SRR393028.2 2/2
CCGATGGCAGCAGACGGCAGGTAGCGGTCCAGCTGCTGGAGCGAGGGCCGGCCACGTATCACAGCCAAGGAGATGA
+
????????????????????????????????????????????????????????????????????????????
@SRR393028.3 3/2
ACAGGAGACGCCGGCCCCGTAGAGCCCTAGGGGCGACGCCACTCCCACTCACTGTCTACTCTCCTCTCACCTCTGC

gzip: stdout: Broken pipe


There is a phred score of 30 which is in the standard range for quality scores.

In [ ]:
#Write about the fastq file here.

There is a phred score of 30 which is in the standard range for quality scores.

# **4. Pre-QC with FastQC**
### *Total Questions: 2*
### *Total Points: 6+6+8 = 20*


---

<div class="alert alert-block alert-warning">
    <h3>4.1 Show your exact command below.</h3>
</div>

In [4]:
#Create a directory for QC results.
!mkdir -p ~/biol6150/ProjectSubmissions/Group8/Project1/fastQCBeforeTrimming
!mkdir -p ~/biol6150/ProjectSubmissions/Group8/Project1/Trimming
!mkdir -p ~/biol6150/ProjectSubmissions/Group8/Project1/fastQCAfterTrimming

In [6]:
#Command here.
#Run fastqc.
!fastqc -o ~/biol6150/ProjectSubmissions/Group8/Project1/fastQCBeforeTrimming/ ~/biol6150/ProjectSubmissions/Group8/Project1/SRR393028_1.fastq.gz ~/biol6150/ProjectSubmissions/Group8/Project1/SRR393028_2.fastq.gz


application/gzip
application/gzip
Started analysis of SRR393028_1.fastq.gz
Approx 5% complete for SRR393028_1.fastq.gz
Approx 10% complete for SRR393028_1.fastq.gz
Approx 15% complete for SRR393028_1.fastq.gz
Approx 20% complete for SRR393028_1.fastq.gz
Approx 25% complete for SRR393028_1.fastq.gz
Approx 30% complete for SRR393028_1.fastq.gz
Approx 35% complete for SRR393028_1.fastq.gz
Approx 40% complete for SRR393028_1.fastq.gz
Approx 45% complete for SRR393028_1.fastq.gz
Approx 50% complete for SRR393028_1.fastq.gz
Approx 55% complete for SRR393028_1.fastq.gz
Approx 60% complete for SRR393028_1.fastq.gz
Approx 65% complete for SRR393028_1.fastq.gz
Approx 70% complete for SRR393028_1.fastq.gz
Approx 75% complete for SRR393028_1.fastq.gz
Approx 80% complete for SRR393028_1.fastq.gz
Approx 85% complete for SRR393028_1.fastq.gz
Approx 90% complete for SRR393028_1.fastq.gz
Approx 95% complete for SRR393028_1.fastq.gz
Analysis complete for SRR393028_1.fastq.gz
Started analysis of SRR39302

<div class="alert alert-block alert-warning">
    <h3>4.2 Show <i>per base sequencing quality</i>?</h3>
    <p>Add a screenshot below</p>
</div>

In [ ]:
##Add image here for 4.2

![Quality_1.png](./Quality_1.png)

![Quality_1.png](./Quality_2.png)

<div class="alert alert-block alert-warning">
    <h3>4.3 Show <i>Per base sequence content</i>?</h3>
    <p>Add a screenshot below</p>
</div>

In [ ]:
##Add image here for 4.3

![Content_1.png](./Content_1.png)

![Content_2.png](./Content_2.png)

# **5. Trimming with fastp or trimmmomatic**
### *Total Questions: 3*
### *Total Points: 10+10+10 = 30*


---

<div class="alert alert-block alert-warning">
    <h3>5.1 What was the tools and command you used?</h3>
    <p>Write the tool name and exact command below</p>
</div>

In [7]:
#extracting fastq.gz files
!gzip -d ~/biol6150/ProjectSubmissions/Group8/Project1/SRR393028_1.fastq.gz
!gzip -d ~/biol6150/ProjectSubmissions/Group8/Project1/SRR393028_2.fastq.gz

In [12]:
#Exact command here

!fastp -i ~/biol6150/ProjectSubmissions/Group8/Project1/SRR393028_1.fastq -I ~/biol6150/ProjectSubmissions/Group8/Project1/SRR393028_2.fastq -o ~/biol6150/ProjectSubmissions/Group8/Project1/Trimming/SRR393028_1.Trimmed.fastq -O ~/biol6150/ProjectSubmissions/Group8/Project1/Trimming/SRR393028_2.Trimmed.fastq -f 9 -t 0


Read1 before filtering:
total reads: 2305520
total bases: 175219520
Q20 bases: 173915740(99.2559%)
Q30 bases: 173915740(99.2559%)

Read2 before filtering:
total reads: 2305520
total bases: 175219520
Q20 bases: 173915740(99.2559%)
Q30 bases: 173915740(99.2559%)

Read1 after filtering:
total reads: 2285934
total bases: 153103227
Q20 bases: 153103227(100%)
Q30 bases: 153103227(100%)

Read2 after filtering:
total reads: 2285934
total bases: 153103227
Q20 bases: 153103227(100%)
Q30 bases: 153103227(100%)

Filtering result:
reads passed filter: 4571868
reads failed due to low quality: 34310
reads failed due to too many N: 4862
reads failed due to too short: 0
reads with adapter trimmed: 65992
bases trimmed due to adapters: 109404

Duplication rate: 0.387461%

Insert size peak (evaluated by paired-end reads): 121

JSON report: fastp.json
HTML report: fastp.html

fastp -i /home/hice1/soliver35/biol6150/ProjectSubmissions/Group8/Project1/SRR393028_1.fastq -I /home/hice1/soliver35/biol6150/Proje

<div class="alert alert-block alert-warning">
    <h3>5.2 How did you perform the trimming?</h3>
    <p>Did you use dynamic built in features or hard coded parameters?</p>
    <p>Did you have to go back and repeat the trimming after run(s) for <b>postQC?</b></p>
</div>

We used the fastp, which is a built in feature of our environment. 

<div class="alert alert-block alert-warning">
    <h3>5.3 What percentage of reads did you lose in the trimming process?</h3>
    <p>5.3.1 Write the total number of reads before and after trimming</p>
    <p>5.3.2 If you had a paired fastq file, are there any reads after trimming present in one file and not the other? (eg: present in reverse but not in forward?)</p>
    
</div>

Before trimming: 4.611040 M
After trimming: 4.571868 M
Percent Lost: 0.85%

# **6. Post-QC with FastQC?**
### *Total Questions: 2*
### *Total Points: 6+8+6=20*


---

<div class="alert alert-block alert-warning">
    <h3>6.1 Show your exact command below.</h3>
</div>

In [2]:
#command here
!fastqc -o ~/biol6150/ProjectSubmissions/Group8/Project1/fastQCAfterTrimming/ ~/biol6150/ProjectSubmissions/Group8/Project1/Trimming/SRR393028_1.Trimmed.fastq ~/biol6150/ProjectSubmissions/Group8/Project1/Trimming/SRR393028_2.Trimmed.fastq

null
null
Started analysis of SRR393028_1.Trimmed.fastq
Approx 5% complete for SRR393028_1.Trimmed.fastq
Approx 10% complete for SRR393028_1.Trimmed.fastq
Approx 15% complete for SRR393028_1.Trimmed.fastq
Approx 20% complete for SRR393028_1.Trimmed.fastq
Approx 25% complete for SRR393028_1.Trimmed.fastq
Approx 30% complete for SRR393028_1.Trimmed.fastq
Approx 35% complete for SRR393028_1.Trimmed.fastq
Approx 40% complete for SRR393028_1.Trimmed.fastq
Approx 45% complete for SRR393028_1.Trimmed.fastq
Approx 50% complete for SRR393028_1.Trimmed.fastq
Approx 55% complete for SRR393028_1.Trimmed.fastq
Approx 60% complete for SRR393028_1.Trimmed.fastq
Approx 65% complete for SRR393028_1.Trimmed.fastq
Approx 70% complete for SRR393028_1.Trimmed.fastq
Approx 75% complete for SRR393028_1.Trimmed.fastq
Approx 80% complete for SRR393028_1.Trimmed.fastq
Approx 85% complete for SRR393028_1.Trimmed.fastq
Approx 90% complete for SRR393028_1.Trimmed.fastq
Approx 95% complete for SRR393028_1.Trimmed.f

<div class="alert alert-block alert-warning">
    <h3>6.2 Show <i>per base sequencing quality</h3>
    <p>If the quality was falling below 20 for certain regions then trim it. Add a screenshot below</p>
</div>

![6.2_picture_1.png](./6.2_picture_1.png)
![6.2_picture_2.png](./6.2_picture_2.png)

Sequence quality after trimming for SRR393028_1 (top) and SRR393028_2 (bottom).

<div class="alert alert-block alert-warning">
    <h3>6.3 Show <i>Per base sequence content</i>?</h3>
    <p>Ideally we shouldn't see adapters at the beginning of the reads in the fastq files. Add a screenshot below</p>
</div>

![6.3_2.png](./6.3_2.png)

After trimming per base sequence content for SRR393028_1

![ContentAfterTrim_2.png](./ContentAfterTrim_2.png)

After trimming per base sequence content for SRR393028_2